### ISSUE TITLE
Incompatible with Jest 26+

### ISSUE THREAD

**Issue by robaw** [mikeborozdin/jest-react-hooks-shallow#34]
When running with Jest 26.6.0 (with Create-React-App v4), all of my tests now fail with the error 'Hooks cannot be defined inside tests. Hook of type "beforeEach" is nested within "Test Name"'. This was mentioned as one of the issues in [this previous issue](https://github.com/mikeborozdin/jest-react-hooks-shallow/issues/18#issuecomment-730303772) , of which the fix was to upgrade to v1.4.2 of jest-react-hooks-shallow. This, however, has not fixed it. Is there some other configuration that is required to make it work with Jest 26.6.0?

Stacktrace:
```
Error: Hooks cannot be defined inside tests. Hook of type "beforeEach" is nested within "toggles tigonMode and saves it in sessionStore".
    at eventHandler (/Users/robwaddell/src/git/string-dashboard/client/node_modules/jest-circus/build/eventHandler.js:105:11)
    at dispatchSync (/Users/robwaddell/src/git/string-dashboard/client/node_modules/jest-circus/build/state.js:65:5)
    at _addHook (/Users/robwaddell/src/git/string-dashboard/client/node_modules/jest-circus/build/index.js:125:27)
    at beforeEach (/Users/robwaddell/src/git/string-dashboard/client/node_modules/jest-circus/build/index.js:135:3)
    at Object.mockUseEffect [as default] (/Users/robwaddell/src/git/string-dashboard/client/node_modules/jest-react-hooks-shallow/src/mock-use-effect/mock-use-effect.ts:14:3)
    at withHooks (/Users/robwaddell/src/git/string-dashboard/client/node_modules/jest-react-hooks-shallow/src/enable-hooks.ts:47:49)
    at Object.<anonymous> (/Users/robwaddell/src/git/string-dashboard/client/src/app/App.test.jsx:54:7)
    at Promise.then.completed (/Users/robwaddell/src/git/string-dashboard/client/node_modules/jest-circus/build/utils.js:276:28)
    at new Promise (<anonymous>)
    at callAsyncCircusFn (/Users/robwaddell/src/git/string-dashboard/client/node_modules/jest-circus/build/utils.js:216:10)
    at _callCircusTest (/Users/robwaddell/src/git/string-dashboard/client/node_modules/jest-circus/build/run.js:212:40)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at _runTest (/Users/robwaddell/src/git/string-dashboard/client/node_modules/jest-circus/build/run.js:149:3)
    at _runTestsForDescribeBlock (/Users/robwaddell/src/git/string-dashboard/client/node_modules/jest-circus/build/run.js:63:9)
    at _runTestsForDescribeBlock (/Users/robwaddell/src/git/string-dashboard/client/node_modules/jest-circus/build/run.js:57:9)
    at _runTestsForDescribeBlock (/Users/robwaddell/src/git/string-dashboard/client/node_modules/jest-circus/build/run.js:57:9)
    at run (/Users/robwaddell/src/git/string-dashboard/client/node_modules/jest-circus/build/run.js:25:3)
    at runAndTransformResultsToJestFormat (/Users/robwaddell/src/git/string-dashboard/client/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapterInit.js:176:21)
```

**Comment by thejuan** [comment]
I'm having the same issue.
I use `enableHooks(jest, { dontMockByDefault: true });` in setup
I'm also using jest-circus as the runner

**Comment by mikeborozdin** [comment]
Hi @robaw and @thejuan ,

Thanks a lot for reporting this! I'm planning to fix it by the end of the year.

**Comment by thejuan** [comment]
I've swapped over to https://github.com/davidmfoley/enzyme-adapter-react-16-with-shallow-effects as my adapter and is working well for Jest 26

**Comment by mdfederici** [comment]
Hi @mikeborozdin. Thanks for the great library. Any update on a fix for this issue?

**Comment by EricKulchycki** [comment]
Hello @mikeborozdin, I was wondering if there's been any progress on this issue as it is blocking me from upgrading react-scripts to the newest version.

**Comment by mikeborozdin** [comment]
Working on the fix. It'll be released by the end of Wednesday (GMT).

**Comment by mikeborozdin** [comment]
@EricKulchycki , @mdfederici , @thejuan , @robaw , and everyone :),

the problem has been resolved - please, upgrade to version `1.5.1`.
